src/tty/test.rs

```diff
@@ -149,6 +149,8 @@ impl Term for DummyTerminal {
         _tab_stop: usize,
         bell_style: BellStyle,
     ) -> DummyTerminal {
+        #[cfg(not(test))]
+        log::error!("Unsupported platform, dummy terminal implementation is being used. Rustyline will not work properly");
```

Comment:
This will only print the error if you have `RUST_LOG=error` set.

Reply:
At least `env_logger` has error-level messages enabled by default. I tested that under WASI the message is printed without any need to set environment variables.

Reply:
What if you don't setup a logger?

Reply:
Then it won't. I preferred a logger, because it's less invasive.

As an alternative, we could change it to `eprintln` and possibly add a feature to silence the warning.